File: python/GeometryBasicsDotProductin3D.py
# Copilot: ignore file
# Kata: Geometry Basics: Dot Product in 3D
# Level: 8kyu
# Solved: 2025-08-22

class Vector:
    def __init__(self, x, y ,z):
        self.x = x  
        self.y = y    
        self.z = z     

def dot_product(a, b):
    return a.x * b.x + a.y * b.y + a.z * b.z  # a⋅b=∣a∣∣b∣cosθ

File: js/GeometryBasicsDotProductin3D.js
// Copilot: ignore file
// Kata: Geometry Basics: Dot Product in 3D
// Level: 7kyu
// Solved: 2025-08-22

class Vector {
    constructor(x, y, z) {
        this.x = x
        this.y = y
        this.z = z
    }
}

function dotProduct(v1, v2) {
    return v1.x * v2.x + v1.y * v2.y + v1.z * v2.z  // a⋅b=∣a∣∣b∣cosθ
}